openwisp_monitoring/device/apps.py

```diff
@@ -109,27 +111,79 @@ def is_working_changed_receiver(cls, instance, is_working, **kwargs):
         device_monitoring = device.monitoring
         initial_status = device_monitoring.status
         status = 'ok' if is_working else 'problem'
+        notification_opts = dict(sender=instance, target=device)
         if not is_working:
-            # Create a related notification explaining why it's not working
-            desc = instance.failure_reason
-            opts = dict(
-                sender=device, level='warning', verb='not working', description=desc
-            )
+            # Select appropriate notification type for reason of not working
+            if 'upgrade' in instance.failure_reason:
+                notification_opts['type'] = 'unreachable_after_upgrade'
+            elif instance.failure_reason:
+                notification_opts['type'] = 'connection_lost'
+            else:
+                notification_opts['type'] = 'device_down'
             if initial_status == 'ok':
                 trigger_device_checks.delay(pk=device.pk)
         else:
             # create a notification that device is working
-            desc = f'{device} has connected successfully.'
-            opts = dict(
-                sender=device,
-                level='info',
-                verb='connected successfully',
-                description=desc,
-            )
+            notification_opts['type'] = 'device_up'
             # if checks exist trigger them else, set status as 'ok'
             if Check.objects.filter(object_id=instance.device.pk).exists():
                 trigger_device_checks.delay(pk=device.pk)
             else:
                 device_monitoring.update_status(status)
-        opts['data'] = {'email_subject': f'[{status.upper()}] {device}'}
-        notify.send(**opts)
+        notify.send(**notification_opts)
+
+    def register_notifcation_types(self):
+        register_notification_type(
+            'device_up',
+            {
+                'name': 'Device Alert',
+                'verb': 'connected successfully',
+                'level': 'info',
+                'email_subject': '[{site.name}] INFO: Device "{notification.target}"',
+                'message': (
+                    '[Device: {notification.target}]({notification.target_link})'
+                    ' has {notification.verb} at {notification.timestamp}.'
```

Comment:
Is `notification.timestamp` a human readable date? Maybe is better to just remove it.